modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/components/export-form/ExportFormModal.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {act, fireEvent, render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import {ExportFormModal} from 'dynamic-data-mapping-form-web/admin/js/components/export-form/openExportFormModal.es';
+import React from 'react';
+
+import '@testing-library/jest-dom/extend-expect';
+
+const props = {
+	csvExport: 'enabled-with-warning',
+	exportFormInstanceURL: 'https://liferay.com/admin/export_form_instance',
+	fileExtensions: {
+		csv: 'CSV',
+		json: 'JSON',
+		xls: 'XLS',
+		xml: 'XML',
+	},
+	portletNamespace: 'portletNamespace_',
+	spritemap: 'spritemap',
+};
+
+describe('ExportFormModal', () => {
+	beforeAll(() => {
+		jest.useFakeTimers();
+	});
+
+	afterAll(() => {
+		jest.useRealTimers();
+	});
+
+	afterEach(() => {
+		jest.clearAllTimers();
+		jest.restoreAllMocks();
+	});
+
+	it('renders', () => {
+		render(<ExportFormModal {...props} />);
+
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		expect(screen.getByText('cancel')).toBeInTheDocument();
+		expect(screen.getByText('export')).toBeInTheDocument();
+		expect(screen.getByText('ok')).toBeInTheDocument();
+		expect(
+			screen.getByText(
+				'the-export-includes-data-from-all-fields-and-form-versions'
+			)
+		).toBeInTheDocument();
+		expect(
+			screen.getByText('timezone-warning-message')
+		).toBeInTheDocument();
+	});
+
+	it('closes modal when clicking the cancel button', () => {
+		render(<ExportFormModal {...props} />);
+
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		userEvent.click(screen.getByText('cancel'));
+
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		expect(screen.queryByText('export')).not.toBeInTheDocument();
+	});
+
+	it('changes warning message when changing file extension', () => {
+		render(<ExportFormModal {...props} />);
+
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		const fileExtension = screen.getByLabelText('file-extension');
+
+		expect(fileExtension).toHaveValue('csv');
+		expect(screen.getByText('csv-warning-message')).toBeInTheDocument();
+
+		userEvent.selectOptions(fileExtension, 'json');
+		fireEvent.change(fileExtension);
```

Comment:
I think we need to call the `fireEvent.change()` due to a bug into the  `selectOptions()` implementation: https://github.com/testing-library/user-event/issues/358

Can you please add a `TODO` in order to remove that after bumping the userEvent version?
```suggestion
		userEvent.selectOptions(fileExtension, 'json');
		
		/* TODO: Remove that after bump userEvent to version 12.0.7 or later https://github.com/testing-library/user-event/issues/358 */
		fireEvent.change(fileExtension);
```